main.go

```diff
@@ -120,6 +126,18 @@ func main() {
 		PprofBindAddress:       pprofAddr,
 	}
 
+	var labelSelectors labels.Selector
+	var err error
+	if watchLabelSelectors != "" {
+		labelSelectors, err = labels.Parse(watchLabelSelectors)
```

Comment:
Note: I like this quite a lot, and would prefer to swap the existing namespace selector parsing to this.